cbindings/p2p.go

```diff
@@ -176,8 +211,16 @@ func P2PdocumentGetAll(nodePtr C.uintptr_t) C.Result {
 }
 
 //export P2PdocumentSync
-func P2PdocumentSync(nodePtr C.uintptr_t, collection *C.char, docIDs *C.char, timeoutStr *C.char) C.Result {
+func P2PdocumentSync(nodePtr C.uintptr_t,
+	collection *C.char,
+	docIDs *C.char,
+	timeoutStr *C.char,
+	identityPtr C.uintptr_t) C.Result {
 	ctx := context.Background()
+	ctx, err := contextWithIdentity(ctx, identityPtr)
+	if err != nil {
+		return returnC(returnGoC(1, err.Error(), ""))
```

Comment:
suggestion: The code seems to do this a lot - perhaps it would be better to write a function for this - this will reduce the noise from the usually unimportant error returns.

e.g.
```
func returnCErr(error) C.Result

func P2PdocumentSync() {
  ...
  if err != nil {
    return returnCErr(err)
  }
}
```